db/queries/statuses/queries.py

```diff
@@ -81,58 +81,70 @@ def update_form_statuses(
         current_form.has_completed = True
     else:
         current_form.status = "IN_PROGRESS"
-    db.session.commit()
 
 
-def update_question_statuses(application_id: str, form_name: str):
+def _is_field_answered(field):
+    answer_or_not_specified = field.get("answer")
+    match answer_or_not_specified:  # noqa
+        case "":
+            return False
+        case []:  # noqa (E211)
+            return False
+        # optional questions return None (not string)
+        # when submitted with no answer
+        case None:
+            return True
+        # default case when there is an answer
+        case _:
+            return True
+
+
+def _determine_answer_status_for_fields(fields_in_question) -> list[bool]:
+    answer_found_list = [_is_field_answered(field) for field in fields_in_question]
+    return answer_found_list
+
+
+def _determine_question_status_from_answers(answer_found_list: list[bool]) -> str:
+
+    # If we found no answers
+    if not answer_found_list or len(answer_found_list) == 0:
+        return "NOT_STARTED"
+    # If all answers are given
+    if all(answer_found_list):
+        return "COMPLETED"
+    # If some answers are given
+    elif any(answer_found_list):
```

Comment:
I like the change to use any, the old implementation was a weird any using all! 😄